org.lflang/src/org/lflang/FileConfig.java

```diff
@@ -805,54 +793,32 @@ public static Path findFile(String fileName, Path directory) {
         // Not found.
         return null;
     }
-     
-     /**
-      * Return the name of the RTI executable.
-      * @return
-      */
-     public String getRTIBinName() {
-         return nameWithoutExtension(srcFile) + RTI_BIN_SUFFIX;
-     }
-     
-     public File getRTIBinFile() {
-         return this.binPath.resolve(getRTIBinName()).toFile();
-     }
-     
-     /**
-      * Return the name of the RTI distribution script.
-      * @return
-      */
-     public String getRTIDistributionScriptName() {
-         return nameWithoutExtension(srcFile) + RTI_DISTRIBUTION_SCRIPT_SUFFIX;
-     }
-     
-     public File getRTIDistributionScriptFile() {
-         return this.binPath.resolve(getRTIDistributionScriptName()).toFile();
-     }
-     
-     public static String nameWithoutExtension(Resource r) throws IOException {
-         return nameWithoutExtension(toPath(r));
-     }
-     
-     /**
-      * Determine which mode the compiler is running in.
-      * Integrated mode means that it is running within an IDE or text editor.
-      * Standalone mode means that it is running on the command line.
-      * 
-      * FIXME: Not sure if that us the right place for this function. But
-      *  the decision which mode we are in depends on a file (the resource),
-      *  thus it seems to fit here.
-      * FIXME: If/when we move away from Xtext, which limits our ability to
-      *  include info in the context about whether we are in Epoch or LSP_FAST,
-      *  it would be preferable not to rely on a method like this that has to
-      *  translate a weird assortment of data into an enum.
-      */
-     public Mode getCompilerMode() {
-         if (context instanceof StandaloneContext) return Mode.STANDALONE;
-         if (resource.getURI().isPlatform()) return Mode.EPOCH;
-         if (context instanceof SlowIntegratedContext) {
-             return ((SlowIntegratedContext) context).getMustBeComplete() ? Mode.LSP_SLOW : Mode.LSP_MEDIUM;
-         }
-         return Mode.LSP_FAST;
-     }
+
+    /**
+     * Return the name of the RTI executable.
+     * @return
+     */
+    public String getRTIBinName() {
+        return nameWithoutExtension(srcFile) + RTI_BIN_SUFFIX;
+    }
+
+    public File getRTIBinFile() {
+        return this.binPath.resolve(getRTIBinName()).toFile();
```

Comment:
These methods also still need some JavaDoc.

Reply:
Thanks for the catch -- I just pushed a commit that adds JavaDoc here. For some reason the diff that appears here on GitHub isn't updating, but the changes are definitely pushed.